controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java

```diff
@@ -299,6 +300,20 @@ public InternalStepRunner(Controller controller) {
             }
 
             throw e;
+        } catch (EndpointCertificateException e) {
+            switch (e.getType()) {
+                case CERT_NOT_AVAILABLE:
+                    // Same as CERTIFICATE_NOT_READY above, only from the controller
+                    Optional<RunStatus> result = startTime.isBefore(controller.clock().instant().minus(Duration.ofHours(1)))
+                            ? Optional.of(deploymentFailed) : Optional.empty();
+                    if (startTime.plus(endpointCertificateTimeout).isBefore(controller.clock().instant())) {
+                        logger.log("Deployment failed to find provisioned endpoint certificate after " + endpointCertificateTimeout);
+                        return Optional.of(RunStatus.endpointCertificateTimeout);
+                    }
+                    return result;
+                default:
+                    throw e; // Should be surfaced / fail deployment
+            }
```

Comment:
Since the cert timeout is lower than the general 1h timeout, this should be simplified :)

Reply:
(Drop computing `result` and just return `Optional.empty()` in its place.